app/views/play/CampaignView.coffee

```diff
@@ -337,23 +341,14 @@ module.exports = class CampaignView extends RootView
     levelPlayCountsRequest.load()
 
   onLoaded: ->
-    if @classroom
-      classroomLevels = @classroom.getLevels()
-      @classroomLevelMap = _.zipObject(classroomLevels.map((l) -> l.get('original')), classroomLevels.models)
-      defaultLanguage = @classroom.get('aceConfig').language
-      for session in @sessions.slice()
-        classroomLevel = @classroomLevelMap[session.get('level').original]
-        if not classroomLevel
-          continue
-        expectedLanguage = classroomLevel.get('primerLanguage') or defaultLanguage
-        if session.get('codeLanguage') isnt expectedLanguage
-          @sessions.remove(session)
-          continue
-    unless @editorMode
-      for session in @sessions.models
-        unless @levelStatusMap[session.get('levelID')] is 'complete'  # Don't overwrite a complete session with an incomplete one
-          @levelStatusMap[session.get('levelID')] = if session.get('state')?.complete then 'complete' else 'started'
-        @levelDifficultyMap[session.get('levelID')] = session.get('state').difficulty if session.get('state')?.difficulty
+    if @isClassroom()
+      @updateClassroomSessions()
+    else
```

Comment:
This if/else means the `unless @editorMode` clause won't be executed when `@isClassroom()` is true but `if @classroom` is false within `updateClassroomSessions`.  Are we ok with that?

Reply:
👍